helpers/depth.py

```diff
@@ -161,15 +162,25 @@ def predict(self, prev_img_cv2, anim_args) -> torch.Tensor:
         
         return depth_tensor
 
-    def save(self, filename: str, depth: torch.Tensor):
+    def save(self, filename: str, depth: torch.Tensor, bit_depth_output):
         depth = depth.cpu().numpy()
         if len(depth.shape) == 2:
             depth = np.expand_dims(depth, axis=0)
         self.depth_min = min(self.depth_min, depth.min())
         self.depth_max = max(self.depth_max, depth.max())
         print(f"  depth min:{depth.min()} max:{depth.max()}")
         denom = max(1e-8, self.depth_max - self.depth_min)
-        temp = rearrange((depth - self.depth_min) / denom * 255, 'c h w -> h w c')
-        temp = repeat(temp, 'h w 1 -> h w c', c=3)
-        Image.fromarray(temp.astype(np.uint8)).save(filename)    
+        if bit_depth_output == 16:
+            temp_image = rearrange((depth - self.depth_min) / denom * 255 * 255, 'c h w -> h w c')
+            temp_image = repeat(temp_image, 'h w 1 -> h w c', c=3)
+            write_png(filename, temp_image.astype(np.uint16));
+        elif bit_depth_output == 32:
+            os.environ["OPENCV_IO_ENABLE_OPENEXR"]="1"
+            temp_image = rearrange((depth - self.depth_min) / denom, 'c h w -> h w c')
+            temp_image = repeat(temp_image, 'h w 1 -> h w c', c=3)
+            cv2.imwrite(filename.replace(".png", ".exr"), temp_image)
+        else: # 8 bit
+            temp_image = rearrange((depth - self.depth_min) / denom * 255, 'c h w -> h w c')
+            temp_image = repeat(temp_image, 'h w 1 -> h w c', c=3)
+            Image.fromarray(temp_image.astype(np.uint8)).save(filename)
```

Comment:
consider either moving this line to same indentation level as `L:183` and `L:184`, or move those two lines to this level + remove the `else`

Reply:
> consider either moving this line to same indentation level as `L:183` and `L:184`, or move those two lines to this level + remove the `else`

This one I didn't understand. There is a three-way option, and `Image...save(filename)` is only used for 8bpc. I think that this one is fine like it is, but maybe I didn't understand your comment.